sensing/pointcloud_preprocessor/src/polygon_remover/polygon_remover.cpp

```diff
@@ -0,0 +1,181 @@
+    marker_.color.g = 1.0;
+    marker_.color.b = 0.0;
+
+    auto make_point = [](float x, float y, float z) {
+      geometry_msgs::msg::Point point;
+      point.x = static_cast<double>(x);
+      point.y = static_cast<double>(y);
+      point.z = static_cast<double>(z);
+      return point;
+    };
+    for (size_t index_cur = 0; index_cur < polygon_cgal_.size(); ++index_cur) {
+      const auto & vertex = polygon_cgal_.at(index_cur);
+
+      // Take the last segment into consideration to connect the loop
+      size_t index_next = index_cur == polygon_cgal_.size() - 1 ? 0UL : index_cur + 1;
+      const auto & vertex_next = polygon_cgal_.at(index_next);
+
+      // Build upper ring
+      auto vertex_up_cur =
+        make_point(static_cast<float>(vertex.x()), static_cast<float>(vertex.y()), 5.0F);
+      auto vertex_up_next =
+        make_point(static_cast<float>(vertex_next.x()), static_cast<float>(vertex_next.y()), 5.0F);
+      marker_.points.emplace_back(vertex_up_cur);
+      marker_.points.emplace_back(vertex_up_next);
+
+      // Build lower ring
+      auto vertex_down_cur =
+        make_point(static_cast<float>(vertex.x()), static_cast<float>(vertex.y()), -5.0F);
+      auto vertex_down_next =
+        make_point(static_cast<float>(vertex_next.x()), static_cast<float>(vertex_next.y()), -5.0F);
+      marker_.points.emplace_back(vertex_down_cur);
+      marker_.points.emplace_back(vertex_down_next);
+
+      // Connect up and down vertices
+      marker_.points.emplace_back(vertex_up_cur);
+      marker_.points.emplace_back(vertex_down_cur);
+    }
+  }
+  polygon_is_initialized_ = true;
+}
+
+sensor_msgs::msg::PointCloud2 PolygonRemoverComponent::remove_updated_polygon_from_cloud(
+  const PointCloud2ConstPtr & cloud_in)
+{
+  if (will_visualize_) {
+    marker_.header.frame_id = cloud_in->header.frame_id;
+  }
+  if (!polygon_is_initialized_) {
+    throw std::runtime_error("Polygon is not initialized. Please use `update_polygon` first.");
+  }
+
+  return remove_polygon_cgal_from_cloud(cloud_in, polygon_cgal_);
+}
+
+sensor_msgs::msg::PointCloud2 PolygonRemoverComponent::remove_polygon_cgal_from_cloud(
+  const sensor_msgs::msg::PointCloud2::ConstSharedPtr & cloud_in_ptr,
+  const std::vector<PointCgal> & polyline_polygon)
+{
+  sensor_msgs::msg::PointCloud2 output;
+  pcl::PointCloud<pcl::PointXYZ> pcl_output;
+
+  sensor_msgs::msg::PointCloud2 transformed_cluster = *cloud_in_ptr;
+
+  for (sensor_msgs::PointCloud2ConstIterator<float> iter_x(transformed_cluster, "x"),
+       iter_y(transformed_cluster, "y"), iter_z(transformed_cluster, "z");
+       iter_x != iter_x.end(); ++iter_x, ++iter_y, ++iter_z) {
+    if (
+      CGAL::bounded_side_2(
+        polyline_polygon.begin(), polyline_polygon.end(), PointCgal(*iter_x, *iter_y), K()) ==
+      CGAL::ON_UNBOUNDED_SIDE) {
+      pcl::PointXYZ p;
+      p.x = *iter_x;
+      p.y = *iter_y;
+      p.z = *iter_z;
+      pcl_output.emplace_back(p);
+    }
```

Comment:
@beginningfan thanks for working on this PR.

Is there a specific reason for using `sensor_msgs::PointCloud2ConstIterator` instead of the `
point_cloud_msg_wrapper::PointCloud2Modifier` and `point_cloud_msg_wrapper::PointCloud2View` from https://gitlab.com/ApexAI/point_cloud_msg_wrapper ?

Reply:
I tried the PointCloud2Modifier and  PointCloud2View in this [way](https://gitlab.com/autowarefoundation/autoware.auto/AutowareAuto/-/blob/master/src/perception/filters/polygon_remover/src/polygon_remover.cpp),  but this algorithm requires the point cloud to have intensity. However, the actual point cloud does not necessarily have intensity information. Is there any different between the algorithm I mentioned above and the algorithm you said

Reply:
@beginningfan I see. This works too of course and I don't expect it to have much of a negative impact in sense of speed.

I think we can think about the tools to modify/access the point clouds once the point types are finalized.

Thanks for your work on this!